Classes/common/HTTP/CDTURLSession.m

```diff
@@ -45,10 +48,21 @@ - (instancetype)initWithDelegate:(NSObject<NSURLSessionDelegate> *)delegate
         _thread = thread;
         _interceptors = [NSArray arrayWithArray:requestInterceptors];
 
-        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+        NSURLSessionConfiguration *config;
+        // Create a unique session id using the address of self.
+        NSString *sessionId = [NSString stringWithFormat:@"com.cloudant.sync.sessionid.%p", self];
+        if ([[NSURLSessionConfiguration class] respondsToSelector:@selector(backgroundSessionConfigurationWithIdentifier:)]) {
```

Comment:
Why is this if/else required? Is this something to do with iOS/OSX versions and API availability? It probably deserves a comment.